src/psyclone/tests/psyir/transformations/chunk_loop_trans_test.py

```diff
@@ -0,0 +1,401 @@
+    incvar = routine.symbol_table.symbol_from_tag(
+            "incvar", symbol_type=DataSymbol, datatype=ScalarType(
+                    ScalarType.Intrinsic.INTEGER,
+                    ScalarType.Precision.SINGLE))
+    parent = Loop()
+    parent.addchild(Reference(lvar))
+    parent.addchild(Literal("10", INTEGER_TYPE))
+    parent.addchild(Reference(incvar))
+    parent.addchild(Schedule())
+    with pytest.raises(TransformationError) as excinfo:
+        chunktrans.validate(parent)
+    assert ("Cannot apply a ChunkLoopTrans to a loop with a non-literal step"
+            " size, but a step expression node of type 'Loop' was found.") \
+        in str(excinfo.value)
+
+
+def test_blockloop_trans_validate2():
+    '''Test the validate method of ChunkLoopTrans for bad step sizes'''
+    chunktrans = ChunkLoopTrans()
+    # Construct a Loop with too large a step-size
+    parent = Loop()
+    parent.addchild(Literal("1", INTEGER_TYPE))
+    parent.addchild(Literal("2560", INTEGER_TYPE))
+    parent.addchild(Literal("128", INTEGER_TYPE))
+    parent.addchild(Schedule())
+    with pytest.raises(TransformationError) as excinfo:
+        chunktrans.validate(parent, {"blocksize": 16})
+    assert ("Cannot apply a ChunkLoopTrans to a loop with larger step size "
+            "(128) than the chosen block size (16).") in str(excinfo.value)
+
+    # Construct a Loop with step-size of 0
+    parent = Loop()
+    parent.addchild(Literal("1", INTEGER_TYPE))
+    parent.addchild(Literal("2560", INTEGER_TYPE))
+    parent.addchild(Literal("0", INTEGER_TYPE))
+    parent.addchild(Schedule())
+    with pytest.raises(TransformationError) as excinfo:
+        chunktrans.validate(parent)
+    assert ("Cannot apply a ChunkLoopTrans to a loop with a step size of 0."
+            in str(excinfo.value))
+
+
+def test_blockloop_trans_validate3():
+    '''Test the validate method of ChunkLoopTrans fails when applying it
+    to a loop which has already been blocked'''
+    chunktrans = ChunkLoopTrans()
+    # Construct a Loop and apply a ChunkLoopTrans to it, then revalidate the
+    # parent loop (can't apply a block loop trans to a block loop trans
+    symbol_table = SymbolTable()
+    parent = Loop()
+    parent.addchild(Literal("1", INTEGER_TYPE))
+    parent.addchild(Literal("512", INTEGER_TYPE))
+    parent.addchild(Literal("1", INTEGER_TYPE))
+    parent.addchild(Schedule())
+    routine = Routine.create("test_routine", symbol_table, [parent])
+    lvar = routine.symbol_table.symbol_from_tag(
+            "lvar", symbol_type=DataSymbol, datatype=ScalarType(
+                    ScalarType.Intrinsic.INTEGER,
+                    ScalarType.Precision.SINGLE))
+    parent._variable = lvar
+    chunktrans.apply(parent)
+#    with pytest.raises(TransformationError) as excinfo:
+#        chunktrans.validate(parent.ancestor(Loop))
+#    assert "Cannot apply a ChunkLoopTrans to an already blocked loop." \
+#           in str(excinfo.value)
```

Comment:
Should this be uncommented or removed?

Reply:
It should be uncommented, I'd commented it out before when i was fixing tests but it seems fine now.